### ISSUE TITLE
Rich text transformation improvement

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#29463]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

A developer feature.

In front ProductController, there is following code you can see below. It deals with transforming a description that comes from backoffice. It alters images a bit in product descriptions and sets some classes for responsivity to them.

I think:
- It's not very good place for it
- It's not reusable.
- It's not used for other descriptions.

```php
    protected function transformDescriptionWithImg($desc)
    {
        $reg = '/\[img\-([0-9]+)\-(left|right)\-([a-zA-Z0-9-_]+)\]/';
        while (preg_match($reg, $desc, $matches)) {
            $link_lmg = $this->context->link->getImageLink($this->product->link_rewrite, $this->product->id . '-' . $matches[1], $matches[3]);
            $class = $matches[2] == 'left' ? 'class="imageFloatLeft"' : 'class="imageFloatRight"';
            $html_img = '<img src="' . $link_lmg . '" alt="" ' . $class . '/>';
            $desc = str_replace($matches[0], $html_img, $desc);
        }

        return $desc;
    }
```

### Describe the solution you'd like

- There is a `filterHtmlContent` hook in ObjectPresenter that deals with transforming all HTML fields. This is a hook where theme modules can hook into and do their job - transform the HTML coming from BO to FO in whatever way they like.
- I think this code should not be in a core, as it's a theme specific code. Where to put it?

### Additional context

Ping @PrestaShop/prestashop-maintainers for specifications.

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by yanmakouf** [comment]
@Hlavtox thanks for your proposition/request. I think it is worth investigating, and, in case, improve. There are so much things that need to be improved, we will look into it and see if it is doeable or not according to the release plan, keep in mind that we cannot introduce bc breaks in a minor version so switching it would imho ( i'm new)  be in the next major (provided it is approved), since moving it would probably induce broken usage we didn't think of.

**Comment by Hlavtox** [comment]
@yanmakouf Yes Yani, no BCs. The question is, what to do with this code? I think this code should be in a theme related module I think. Or just removed from the default theme, because it can be styled with CSS.

**Comment by sLorenzini** [comment]
Hello @Hlavtox,

Thank you for your suggestion. The Maintainer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. 
We’re waiting for your PR 🚀

Thank you
